src/core/execute.c

```diff
@@ -4270,8 +4270,16 @@ static int exec_child(
                         }
                 }
 
-                /* This is done before enforce_user, but ambient set
-                 * does not survive over setresuid() if keep_caps is not set. */
+                /* Ambient capabilities are cleared during setresuid (in enforce_user) even with
+                 * keep-caps set.
+                 * To be able to raise the ambient capabilities after setresuid they have to be
+                 * added to the inherited set and keep caps has to be set (done in enforce_user).
+                 * After setresuid the ambient capabilities can be raised as they are present in
+                 * the permitted and inhertiable set. However it is possible that someone wants to
+                 * set ambient capabilities without changing the user, so we also set the ambient
+                 * capabilities here.
+                 * The requested ambient capabilities are raised in the inheritable set if the
+                 * second argument is true. */
```

Comment:
hmm, it's customary in our codebase comments to suffix function names with () to indicate they are functions. i.e. setresuid → setresuid(), enforce_user → enforce_user() and so on.